src/main/java/org/example/WikipediaSearch.java

```diff
@@ -0,0 +1,78 @@
+package org.example;
+
+import com.google.gson.*;
+import okhttp3.HttpUrl;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+import java.io.IOException;
+import java.net.URLEncoder;
+
+class WikipediaSearch {
+    private static final String WIKIPEDIA_API_BASE_URL = "https://en.wikipedia.org/w/api.php";
+
+    // this class uses OkHttp client to create an HTTP request to the API endpoint and retrieve the search results
+    private static final OkHttpClient client = new OkHttpClient();
+    private static final Gson gson = new GsonBuilder().setPrettyPrinting().create();
+
+    /**
+     * @search() this method searches for a topic by making an Api request to the wikipedia API endpoint.
+     * @param args passing the topic as arguments "topic:oman" topic[0] split by ':' and oman[1]
+     */
+    public void search(String[] args) {
+
+        try {
+            /**
+             * URL encoding converts characters into a format that can be transmitted over the Internet.
+             * URLs can only be sent over the Internet using the ASCII character-set.
+             * Since URLs often contain characters outside the ASCII set, the URL has to be converted into a valid ASCII format
+             * URL encoding replaces unsafe ASCII characters with a "%" followed by two hexadecimal digits
+             */
+            String encodedTopic = URLEncoder.encode(args[0].split(":")[1], "UTF-8");
+
+            HttpUrl.Builder urlBuilder = HttpUrl.parse(WIKIPEDIA_API_BASE_URL).newBuilder();
+            urlBuilder.addQueryParameter("action", "query");
+            urlBuilder.addQueryParameter("format", "json");
+            urlBuilder.addQueryParameter("list", "search");
+            urlBuilder.addQueryParameter("srsearch", encodedTopic);
+
+            Request request = new Request.Builder()
+                    .url(urlBuilder.build())
+                    .get()
+                    .build();
+
+            try (Response response = client.newCall(request).execute()) {
+
+                if (response.isSuccessful()) {
+                    // Extract the response body as a String
+                    String responseBody = response.body().string();
+
+                    // Using a JSON parsing library allows the code to easily extract the relevant data from the API response and work with it in a structured way.
+                    JsonObject responseJson = JsonParser.parseString(responseBody).getAsJsonObject();
```

Comment:
Please mention the comment explaining how this JsonObject JsonArray and JsonElement works together.

Reply:
done